joern-cli/src/main/scala/io/joern/joerncli/JoernExport.scala

```diff
@@ -156,13 +159,46 @@ object JoernExport extends App {
   /** for each method in the cpg: recursively traverse all AST edges to get the subgraph of nodes within this method add
     * the method and this subgraph to the export add all edges between all of these nodes to the export
     */
-  private def splitByMethod(cpg: Cpg): IterableOnce[SubGraph] = {
+  private def splitByMethod(cpg: Cpg): IterableOnce[MethodSubGraph] = {
     cpg.method.map { method =>
-      val sanitizedMethodName = method.name.replaceAll("[^a-zA-Z0-9-_\\.]", "_")
-      SubGraph(name = s"${method.filename}/$sanitizedMethodName", nodes = method.ast.toSet)
+      MethodSubGraph(methodName = method.name, methodFilename = method.filename, nodes = method.ast.toSet)
     }
   }
 
+  /** @param windowsFilenameDeduplicationHelper
+    *   utility map to ensure we don't override output files for identical method names
+    */
+  private def sanitizedFileName(
+    methodName: String,
+    methodFilename: String,
+    fileExtension: String,
+    windowsFilenameDeduplicationHelper: mutable.Set[String]
+  ): String = {
+    val sanitizedMethodName = methodName.replaceAll("[^a-zA-Z0-9-_\\.]", "_")
+    val isWindows           = System.getProperty("os.name").toLowerCase.contains("win")
```

Comment:
There is `scala.util.Properties.isWin` for this.

Reply:
nice, thanks!